src/chips/nrf51822/gpio.rs

```diff
@@ -55,19 +127,54 @@ impl hil::gpio::GPIOPin for GPIOPin {
     }
 
     fn toggle(&self) {
-        unimplemented!();
+        // TODO: check need for a atomic XOR operator
+        GPIO().out.set((1 << self.pin) ^ GPIO().out.get());
     }
 
     fn read(&self) -> bool {
-        unimplemented!();
+        GPIO().in_.get() & (1 << self.pin) != 0
     }
 
-    fn enable_interrupt(&self, client_data: usize, mode: hil::gpio::InterruptMode) {
-        unimplemented!();
+    fn enable_interrupt(&self, _client_data: usize, _mode: hil::gpio::InterruptMode) {
+       self.client_data.set(_client_data);
+       let mut mode_bits: u32 = 1; // Event
+       mode_bits |= match _mode {
```

Comment:
underscore prefix implies in Rust that the argument isn't used (in fact, the compiler ignores these arguments when warning for unused variables).